web/server/go.md

```diff
@@ -0,0 +1,498 @@
+# Web Applications in Go
+
+This section is meant to contain of some of the lessons
+Truss has had with building web applications in Go.
+It borrows from the format of [Effective Go](https://golang.org/doc/effective_go.html)
+along with some lightweight "starter templates",
+to provide some basic standards and patterns we follow.
+Code samples are not meant to be "copy and paste" ready,
+but instead provide fundamental principles
+that can be incorporated into many applications, frameworks, and packages.
+
+We attempt to use plain Go code when possible,
+but package code may be incorporated or pseudocoded
+to mimic production use cases.
+
+## Application Layers
+
+Defining application layers is crucial
+to a maintainable API or web service.
+Picking responsibilities and contracts across layers
+can define how an application grows
+and how fast code iterations in a mature codebase will take.
+In Go,
+these layers will likely take form as packages.
+There is no one-size-fits-all approach to defining the contract of these packages,
+but we'll attempt to define some common patterns and inform future approaches.
+
+### Handlers
+
+The term "handler" is used to describe the layer that accepts HTTP requests
+and returns a response.
+While it may vary in implementation between different libraries,
+the term comes from the Go standard library [HTTP Package](https://golang.org/pkg/net/http/#Handler).
+
+The responsibility of a handler is to accept an HTTP request,
```

Comment:
Yes to all of this.

Also, I tend to think of the `Handler`s being in the "Transport Layer". It's main responsibility is to translate from transport-native "language" to internally-used "language" used in the service layer: JSON -> biz/model objects; cookies/headers -> principal/user object; etc...